src/components/FilterHeaderCategories.js

```diff
@@ -0,0 +1,112 @@
+// @flow
+import React from "react";
+import { View, StyleSheet, TouchableHighlight, Image } from "react-native";
+import type { StyleObj } from "react-native/Libraries/StyleSheet/StyleSheetTypes";
+import {
+  interestButtonBgColor,
+  interestButtonTextColor,
+  filterShowMeTextColor,
+  categoriesFilterButtonBgColor
+} from "../constants/colors";
+import text from "../constants/text";
+import Text from "./Text";
+import CategoriesPills from "./CategoriesPills";
+import chevronRightImg from "../../assets/images/chevronRight.png";
+
+type Props = {
+  selectedCategories: Set<string>,
+  onFilterPress: Function
+};
+
+type CategoriesFilterButtonProps = {
+  style?: StyleObj,
+  onPress: Function
+};
+
+const CategoriesFilterButton = ({
+  style,
+  onPress
+}: CategoriesFilterButtonProps) => (
+  <TouchableHighlight
+    accessibilityTraits={["button"]}
+    accessibilityComponentType="button"
+    style={[styles.categoriesFilterButton, style]}
+    onPress={onPress}
+  >
+    <Image source={chevronRightImg} />
+  </TouchableHighlight>
+);
+
+CategoriesFilterButton.defaultProps = {
+  style: {}
+};
+
+const FilterHeaderCategories = ({ selectedCategories, onFilterPress }: Props) =>
+  selectedCategories.size === 0 ? (
+    <View style={styles.contentInterest}>
+      <Text type="h1" style={styles.filterTitle}>
+        {text.filterTitle}
+      </Text>
+      <View style={styles.interestButton}>
+        <Text type="h2" style={styles.interestButtonText}>
+          {text.filterByInterest}
+        </Text>
+        <CategoriesFilterButton onPress={onFilterPress} />
+      </View>
+    </View>
+  ) : (
+    <View style={styles.categoryPillsContainer}>
+      <CategoriesPills
+        style={styles.categoryPills}
+        selectedCategories={selectedCategories}
+      />
+      <CategoriesFilterButton
+        style={styles.categoriesFilterOverButton}
+        onPress={onFilterPress}
+      />
+    </View>
+  );
+
+const styles = StyleSheet.create({
+  contentInterest: {
+    alignItems: "center",
+    flexDirection: "row"
+  },
+  filterTitle: {
+    color: filterShowMeTextColor,
+    paddingTop: 5,
+    marginRight: 8
+  },
+  categoryPillsContainer: {
+    flexDirection: "row"
+  },
+  categoriesFilterButton: {
+    width: 38,
+    height: 40,
```

Comment:
Because we can't use a fixed height for the `interestButton`, this might need a height of 100%? Also it should have a minimal width of 44, which is the minimal size for touch areas for accessibility reasons.

I think I saw something about making only the touch area larger, but not the view itself. This might help us to still use the designed sizes.

Reply:
According to https://facebook.github.io/react-native/docs/view.html#hitslop the touch area would not extend outside of the parent, so we need to bump the design to 44.

Reply:
Ah, so there's a catch. So if UX wants this to be of height 40, we can still use our `Touchable` component and override the `minHeight` for it.

Reply:
Just updated the design to 44 and had a review with ux.

Reply:
Awesome :+1: